x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -573,43 +615,77 @@ void merge(MergeableIndicesPrivilege other) {
             }
         }
 
+        private static void collatePrivilegesByRemoteIndices(
+            final RemoteIndicesPrivileges[] remoteIndicesPrivileges,
+            final boolean allowsRestrictedIndices,
+            final Map<Set<String>, Map<Set<String>, MergeableIndicesPrivilege>> remoteIndicesPrivilegesMap
+        ) {
+            assert remoteIndicesPrivileges != null;
+            // if a remote index privilege is an explicit denial, then we treat it as non-existent to stay consistent with local index
+            // privileges
+            final boolean isExplicitDenial = remoteIndicesPrivileges.length == 1
+                && "none".equalsIgnoreCase(remoteIndicesPrivileges[0].indicesPrivileges().getPrivileges()[0]);
```

Comment:
Move "none" to a constant?

Reply:
Does "none" have to be the first and only entry in the privileges list, or can it be in other positions?

Also, I had assumed empty list of privileges implied none, so I am a little confused why "none" is needed.

Reply:
This is essentially a copy of existing code for indices privileges. Check out [collatePrivilegesByIndices](https://github.com/elastic/elasticsearch/pull/90614/files#diff-65dc248b6cd87fca037bd9e993b2aab3828578878ab92e507a88ce89925b493eR641). 

Based on the existing logic, `none` has to be the first and only entry for this to kick in. An empty list of privileges does imply none but we're handling end-users specifying explicit denial here. I'm staying consistent here with what we do for non-remote index privileges, and what we do for non-remote index privileges we do to stay consistent with past behavior (the code comment from before my change indicates: "if a index privilege is an explicit denial, then we treat it as non-existent since we skipped these in the past when merging")

Reply:
I don't feel strongly. We can defer it.